src/Forms/GridField/GridFieldSudoMode.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace SilverStripe\Forms\GridField;
+
+use SilverStripe\Forms\FormField;
+use SilverStripe\Forms\SudoModePasswordField;
+use SilverStripe\View\ArrayData;
+use SilverStripe\View\SSViewer;
+
+/**
+ * GridFieldSudoMode warning that the managed data is protected by SudoMode which will
+ * need to be activated before the data can be edited.
+ *
+ * This is particularly important for GridFields that contain no data, which provide
+ * no other way to activate sudo mode.
+ */
+class GridFieldSudoMode extends AbstractGridFieldComponent implements GridField_HTMLProvider
+{
+    private string $sectionTitle;
+
+    private int $columnCount;
+
+    public function __construct(string $sectionTile, int $columnCount)
+    {
+        $this->sectionTitle = $sectionTile;
+        $this->columnCount = $columnCount;
+    }
+
+    public function getHTMLFragments($gridField)
+    {
+        $form = $gridField->getForm();
+        // Do not render this component if the entire form is protected by sudo mode.
+        // If the form is protected then there will already be a SudoModePasswordField as the first form field
+        if ($form->getFormRequiresSudoMode()) {
+            return [];
+        }
+        $field = SudoModePasswordField::create();
+        $field->setCollapsed(true);
+        $field->setSectionTitle($this->sectionTitle);
+        // Form needs to be set otherwise the field will not render correctly
+        $field->setForm($form);
+        // Should use silverstripe/admin/themes/cms-forms/templates/SilverStripe/Forms/FormField_holder.ss
+        // The lowercase "e" in "extraClass" is necessary to match the template
+        // The is necessary for the logic in SudoModePasswordFieldEntwine.js to initialise the component
+        $holderTemplateData = ArrayData::create([
+            'extraClass' => $field->extraClass(),
+            'HolderID' => $field->HolderID(),
+            'Field' => $field,
+        ]);
+        $holderTemplate = SSViewer::get_templates_by_class(FormField::class, '_holder');
+        $fieldHolder = $holderTemplateData->renderWith($holderTemplate);
+        $templateData = ArrayData::create([
+            'ColumnCount' => $this->columnCount,
+            'FieldHolder' => $fieldHolder,
+        ]);
+        $template = SSViewer::get_templates_by_class($this, '', __CLASS__);
+        return [
+            'header' => $templateData->renderWith(
+                $template,
+                ['Colspan' => count($gridField->getColumns() ?? [])]
```

Comment:
Why are we setting `ColSpan` here instead of as part of the `ArrayData`?

Reply:
Removed